test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Formatter/DeltaTests.cs

```diff
@@ -412,6 +412,28 @@ public async Task PatchShouldSupportNonSettableCollectionProperties()
             Assert.Equal(3, query.Addresses.Count);
             Assert.Equal(3, query.Orders.Count);
         }
+
+        [Fact]
+        public async Task PatchShouldSupportComplexDerivedTypeTransform()
+        {
+            HttpRequestMessage patch = new HttpRequestMessage(new HttpMethod("MERGE"), BaseAddress + "/odata/DeltaCustomers(6)");
+            dynamic data = new ExpandoObject();
+            data.Addresses = Enumerable.Range(10, 3).Select(i => new DeltaAddress { ZipCode = i });
+             
```

Comment:
The above test cases update the (6) customers, and you do the similar thing for same customers.

In the controller, it's a static List. Any concurrency problem?